pkg/restapi/logspec/operation/operations.go

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package operation
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/trustbloc/edge-core/pkg/log"
+
+	"github.com/trustbloc/edge-service/pkg/internal/common/support"
+	commhttp "github.com/trustbloc/edge-service/pkg/restapi/internal/common/http"
+	"github.com/trustbloc/edge-service/pkg/restapi/model"
+)
+
+const (
+	logSpecEndpoint = "/logspec"
+
+	invalidLogSpec = `Invalid log spec. It needs to be in the following format: ` +
+		`ModuleName1=Level1:ModuleName2=Level2:ModuleNameN=LevelN:AllOtherModuleDefaultLevel
+Valid log levels: critical,error,warn,info,debug
+Error: %s`
+
+	multipleDefaultValues = "multiple default values found"
+
+	getLogSpecPrepareErrMsg = "Failure while preparing log level response: %s."
+)
+
+// Handler represents an HTTP handler for each controller API endpoint
+type Handler interface {
+	Path() string
+	Method() string
+	Handle() http.HandlerFunc
+}
+
+// StringBuilder represents a type that can build up a string to be used as the response for a get log spec call.
+type StringBuilder interface {
+	Write(p []byte) (int, error)
+	String() string
+	Reset()
+}
+
+type moduleLevelPair struct {
+	module   string
+	logLevel log.Level
+}
+
+// GetRESTHandlers gets all controller API handlers available for this service
+func GetRESTHandlers() []Handler {
+	return []Handler{
+		support.NewHTTPHandler(logSpecEndpoint, http.MethodPut, logSpecPutHandler),
+		support.NewHTTPHandler(logSpecEndpoint, http.MethodGet, logSpecGetHandler),
+	}
+}
+
+// Change Log Specification swagger:route PUT /logspec changeLogSpecReq
+//
+// Changes the current log specification.
+// Format: ModuleName1=Level1:ModuleName2=Level2:ModuleNameN=LevelN:AllOtherModuleDefaultLevel
+// Valid log levels: critical,error,warn,info,debug
+//
+// Note that this will not work properly if a module name contains an '=' character.
+//
+// Responses:
+//    default: genericError
+//        200: emptyRes
+func logSpecPutHandler(rw http.ResponseWriter, req *http.Request) {
+	changeLogSpec(rw, req)
+}
+
+// Get Current Log Specification swagger:route GET /logspec getLogSpecReq
```

Comment:
same here `/logspec your_name changeLogSpecReq`

Reply:
That was intentional - in this case, it's shared between multiple services, so there's no one name that really fits it. It shows up under "default" in the OpenAPI UI, which I think is appropriate.

Also, I'll be moving this to edge-core since it's repeated in a lot of projects and very generic.